www/scripts/productlist/ListOfProducts.js

```diff
@@ -213,20 +214,37 @@ function (declare, domClass, domConstruct, ItemFileWriteStore, topic,
         + name[0].toUpperCase()
         + '</span></div>';
 
-      if (!item.connected || !item.accessible) {
+      var dbStatus = item.databaseStatus;
+
+      var dbStatusMessages = {};
+      dbStatusMessages[DBStatus.OK] = 'Database is up to date.';
+      dbStatusMessages[DBStatus.MISSING] = 'Database is missing.';
+      dbStatusMessages[DBStatus.FAILED_TO_CONNECT] = 'Failed to connect to the database.';
+      dbStatusMessages[DBStatus.SCHEMA_MISMATCH_OK] = 'Schema mismatch: migration is possible.';
+      dbStatusMessages[DBStatus.SCHEMA_MISMATCH_NO] = 'Schema mismatch: migration not available.';
+      dbStatusMessages[DBStatus.SCHEMA_MISSING] = 'Schema is missing.';
+      dbStatusMessages[DBStatus.SCHEMA_INIT_ERROR] = 'Schema initialization error.';
+
+      dbStatusMsg = dbStatusMessages[dbStatus];
+
+      if (dbStatusMsg === undefined){
+         dbStatusMsg = "Unknown database error";
+      }
+
+      if (dbStatus !== 0 || !item.accessible) {
         name = '<span class="product-error">'
           + name + '</span>';
 
-        if (!item.connected) {
-          statusIcon = '<span class="customIcon product-error"></span>';
-          description = '<span class="product-description-error database">'
-            + 'The database connection for this product could not be made!'
-            + '</span><br />' + description;
-        } else if (!item.accessible) {
+        if (!item.accessible) {
           statusIcon = '<span class="customIcon product-noaccess"></span>';
           description = '<span class="product-description-error access">'
             + 'You do not have access to this product!'
             + '</span><br />' + description;
+        } else if (dbStatus !== 0) {
+          statusIcon = '<span class="customIcon product-error"></span>';
+          description = '<span class="product-description-error database">'
+            + dbStatusMsg + ' (use server command for schema upgrades)' +
```

Comment:
Remove `+` from the end of line. This will remove `NaN` string from the description.
![cc_products_error](https://user-images.githubusercontent.com/6695818/33545533-ce7ddc36-d8de-11e7-8e54-398d5d0fb6d0.png)